xarray/core/datatree.py

```diff
@@ -463,8 +463,10 @@ def __init__(
 
         super().__init__(name=name)
         self._set_node_data(_coerce_to_dataset(data))
-        self.parent = parent
-        self.children = children
+
+        # shallow copy to avoid modifying arguments in-place (see GH issue #9196)
+        self.parent = parent.copy() if parent is not None else None
```

Comment:
Can we remove `parent` from the constructor instead?

It's a little weird to only be able to set the _direct_ parent (and corresponding sibling), but not more distant relatives.

Reply:
Yeah I was going to do that in a follow-up PR, but I can do it in this one instead.